core/src/main/scala/cats/instances/list.scala

```diff
@@ -82,6 +82,10 @@ trait ListInstances extends cats.kernel.instances.ListInstances {
 
       override def foldM[G[_], A, B](fa: List[A], z: B)(f: (B, A) => G[B])(implicit G: Monad[G]): G[B] =
         Foldable.iteratorFoldM(fa.toIterator, z)(f)
+
+      override def fold[A](fa: List[A])(implicit A: Monoid[A]): A = A.combineAll(fa)
```

Comment:
I was confused about that too.

When you run the [`MapMonoidBench`](https://github.com/typelevel/cats/blob/master/bench/src/main/scala/cats/bench/MapMonoidBench.scala#L24-L30), you can see that the first using `Monoid.combineAll` is considerably faster than the second.

That's because `Monoid.combineAll` is overridden in some instances, eg [`Map`](https://github.com/typelevel/cats/blob/master/kernel/src/main/scala/cats/kernel/instances/map.scala#L41-L51) and [more general `TraversableOnce`](https://github.com/typelevel/cats/blob/master/kernel/src/main/scala/cats/kernel/instances/StaticMethods.scala#L78-L81).